src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -347,18 +363,21 @@ def updateFromPlugin(self):
         self.parent.communicate.statusBarUpdateSignal.emit(msg)
         logging.info(msg)
 
-    def checkModel(self, model_str):
+    def checkModel(self, full_path):
         """
-        Run the ast check
-        and return True if the model is good.
-        False otherwise.
+        Run ast and model checks
+        Attempt to return the line number of the error if any
+        :param full_path: full path to the model file
         """
         # successfulCheck = True
         error_line = 0
         try:
+            with open(full_path, 'r', encoding="utf-8") as plugin:
+                model_str = plugin.read()
             ast.parse(model_str)
+            GuiUtils.checkModel(full_path)
 
-        except SyntaxError as ex:
+        except Exception as ex:
             msg = "Error building model: " + str(ex)
             logging.error(msg)
```

Comment:
The whole Exception handler here grew quite large - please consider refactoring it into a separate method, so the exception flow is clear.